presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/ExpressionVerifier.java

```diff
@@ -74,7 +79,31 @@ protected Boolean visitInPredicate(InPredicate actual, Expression expectedExpres
     {
         if (expectedExpression instanceof InPredicate) {
             InPredicate expected = (InPredicate) expectedExpression;
-            return process(actual.getValue(), expected.getValue()) && process(actual.getValueList(), expected.getValueList());
+
+            if (actual.getValueList() instanceof InListExpression) {
+                return process(actual.getValue(), expected.getValue()) && process(actual.getValueList(), expected.getValueList());
+            }
+            else {
+                checkState(
+                        expected.getValueList() instanceof InListExpression,
+                        "I didn't expect the expected value list to be unpacked. Time to update the DSL!");
+                /*
+                 * If the expected value is a value list, but the actual is e.g. a SymbolReference,
+                 * we need to unpack the value from the list so that when we hit visitSymbolReference, the
+                 * expected.toString() call returns something that the expressionAliases actually contains.
+                 * For example, InL istExpression.toList returns "(onlyitem)" rather than "onlyitem".
+                 *
+                 * This is required because actual passes through the analyzer, planner, and possibly optimizers,
+                 * one of which sometimes takes the liberty of unpacking the InListExpression.
+                 *
+                 * Since the expected value doesn't go through all of that, we have to deal with the case
+                 * of the actual value being unpacked, but the expected value being an InListExpression.
+                 */
+                List<Expression> values = ((InListExpression) expected.getValueList()).getValues();
```

Comment:
Maybe this would be clearer if you extract the unpacking logic to a method? It's not so confusing/radical as the long comment would suggest.